reporting/reports.py

```diff
@@ -115,7 +116,7 @@ def _set(self, tag, value):
                         # joined together and the second one is column
                         # whose value should be used to distinguish if
                         # we still have the same entity or not.
-                        for l in filter(lambda x: x != '', re.split('\n', value)):
+                        for l in [x for x in re.split('\n', value) if x != '']:
```

Comment:
Not sure why you have to make it slower. The `lambda` here is simply not needed:

```python
for l in filter(None, value.split(os.linesep)):
```

Comprehensions are usually slower than `filter` (0.570 usec vs 0.500 usec).

Reply:
Here I'm more in favor of using list comprehensions than using `filter` + `lambda`. Performance-wise they are both similar, actually list comprehensions run even faster here on my tests.

There are also some readability recommendations in favor of list comprehensions: https://docs.quantifiedcode.com/python-anti-patterns/readability/using_map_or_filter_where_list_comprehension_is_possible.html

Reply:
@meaksh I certainly would not refer to a suggestive doc from startup in Berlin. :wink: The map/lambda was a perfectly a pattern for decades and GvR first wanted to kill lambdas at all, then [this tweet](https://twitter.com/gvanrossum/status/391769557758521345?lang=en)... I mean, this is very debatable.

In any case, if you want to go with LC, then at least make them generators so you won't suck into the memory the entire thing just to iterate over it. And yes, map/filter in generally is way faster than LC.